src/metabase/search/ingestion.clj

```diff
@@ -115,25 +115,45 @@
   false)
 
 (defn update!
-  "Update all active engines' existing indexes with the given documents"
-  [documents-reducible]
-  (when-let [engines (seq (search.engine/active-engines))]
-    (if (= 1 (count engines))
-      (search.engine/update! (first engines) documents-reducible)
-      ;; TODO um, multiplexing over the reducible awkwardly feels strange. We at least use a magic number for now.
-      (doseq [batch (eduction (partition-all 150) documents-reducible)
-              e     engines]
-        (search.engine/update! e batch)))))
+  "Update all active engines' existing indexes with the given documents. Passed remove-documents will be deleted from the index."
+  [documents-reducible removed-models-reducible]
+  (doseq [e (seq (search.engine/active-engines))]
+    (u/prog1
+      (search.engine/update! e documents-reducible)
+      (doseq [batch (eduction (partition-all 1000) removed-models-reducible)]
+        (doseq [[group ids] (u/group-by first second batch)]
+          (search.engine/delete! e group ids))))))
+
+(defn- extract-model-and-id
```

Comment:
I think it's worth a clear comment here that were're not supporting compound primary keys (yet)

Reply:
And that the effective behavior will be to not try to delete the corresponding models.